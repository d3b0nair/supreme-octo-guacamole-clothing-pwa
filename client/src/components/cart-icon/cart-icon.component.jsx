import React from "react";
import { connect } from "react-redux";
import { toggleCartHidden } from "../../redux/cart/cart.actions";
import { selectCartItemsCount } from "../../redux/cart/cart.selector";
import { createStructuredSelector } from "reselect";

import {
  CartIconWrapper,
  ShoppingIconWrapper,
  ItemCountWrapper,
} from "./cart-icon.styles";

function CartIcon({ toggleCartHidden, itemCount }) {
  return (
    <CartIconWrapper onClick={toggleCartHidden}>
      <ShoppingIconWrapper />
      <ItemCountWrapper>{itemCount}</ItemCountWrapper>
    </CartIconWrapper>
  );
}
const mapDispatchToProps = (dispatch) => ({
  toggleCartHidden: () => dispatch(toggleCartHidden()),
});

const mapStateToProps = createStructuredSelector({
  itemCount: selectCartItemsCount,
});

export default connect(mapStateToProps, mapDispatchToProps)(CartIcon);
