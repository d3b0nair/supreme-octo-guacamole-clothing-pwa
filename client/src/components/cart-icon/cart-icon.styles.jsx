import styled from "styled-components";

import { ReactComponent as ShoppingIcon } from "../../assets/shopping-cart.svg";
const animationSpeed = `cubic-bezier(0.25, 0.45, 0.45, 0.95)`;
export const CartIconWrapper = styled.div`
  width: 45px;
  height: 45px;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  @media screen and (max-width: 800px) {
    width: 55px;
    height: 55px;
  }
  @media screen and (max-width: 300px) {
    width: 55px;
    height: 55px;
  }
  &:hover {
    transform: scale(1.25);
    color: #ff6e7f;
    transition: transform 0.6s ${animationSpeed};
  }
  &:hover path {
    fill: #ff6e7f !important;
    transition: fill 0.3s ${animationSpeed};
  }
  &:hover span {
    color: white;
    transition: color 0.6s ${animationSpeed};
  }
`;
CartIconWrapper.displayName = "CartIconWrapper";
export const ShoppingIconWrapper = styled(ShoppingIcon)`
  width: 46px;
  height: 46px;
`;

export const ItemCountWrapper = styled.span`
  color: #77a1d3;
  position: absolute;
  font-size: 15px;
  font-weight: bold;
  bottom: 8px;
  @media screen and (max-width: 800px) {
    bottom: 14px;
  }
`;
