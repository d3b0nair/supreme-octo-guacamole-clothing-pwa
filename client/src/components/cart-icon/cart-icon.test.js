import React from "react";
import {
  render,
  fireEvent,
  waitFor,
  screen,
  cleanup,
} from "@testing-library/react";
import { CartIcon } from "./cart-icon.component";

const mockToggleCartHidden = jest.fn();
const mockItemCount = 0;
const mockProps = (otherProps) => {
  return {
    ...otherProps,
    itemCount: mockItemCount,
    toggleCartHidden: mockToggleCartHidden,
  };
};
const component = (mockProps) => {
  return <CartIcon {...mockProps} />;
};

afterEach(cleanup);

describe("CartIcon component", () => {
  it("should mount", () => {
    const { container } = render(component(mockProps()));
    expect(container).toMatchSnapshot();
  });
  it("should call toggleCartHidden on a button click", async () => {
    render(component(mockProps()));
    await waitFor(() => fireEvent.click(screen.getByText(mockItemCount)));
    expect(mockToggleCartHidden).toHaveBeenCalled();
  });
});
