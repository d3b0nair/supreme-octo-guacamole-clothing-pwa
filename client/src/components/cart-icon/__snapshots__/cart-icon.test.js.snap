// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component should mount 1`] = `
<div>
  <div
    class="sc-bdvvtL jTehkV"
  >
    <svg
      class="sc-gsDKAQ tKSVz"
    >
      shopping-cart.svg
    </svg>
    <span
      class="sc-dkPtRN hSsJJR"
    >
      0
    </span>
  </div>
</div>
`;
